workflow-controller/controller-service/src/main/java/io/hyscale/controller/directive/ServiceSpecDirectiveUpdateHandler.java

```diff
@@ -0,0 +1,167 @@
+
+	/**
+	 * Performs inplace update on passed object node
+	 * 
+	 * Implementation: Extending class provides the model of objectNode, update
+	 * function and path from the parent
+	 * 
+	 * @param serviceSpecObjNode - this will be updated
+	 */
+	public void update(ObjectNode serviceSpecObjNode) throws HyscaleException {
+
+		T t = getModel(serviceSpecObjNode);
+		if (t == null) {
+			return;
+		}
+		int initialHash = t.hashCode();
+		t = updateObject(t);
+		int updatedHash = t.hashCode();
+
+		if (initialHash != updatedHash) {
+			updateDirective(serviceSpecObjNode, t);
+		}
+	}
+
+	/**
+	 * Replaces updated object in serviceSpecObject at path provided by
+	 * {@link #getPath()}
+	 * 
+	 * @param serviceSpecObject
+	 * @param updatedObject
+	 */
+	protected void updateDirective(ObjectNode serviceSpecObject, Object updatedObject) {
+		if (serviceSpecObject == null) {
+			return;
+		}
+		String[] path = getPath();
+		// get parent node
+		JsonNode jsonNode = getNodeAtPath(serviceSpecObject, Arrays.copyOfRange(path, 0, path.length - 1));
+		if (jsonNode == null) {
+			return;
+		}
+		String field = path[path.length - 1];
+		if (jsonNode.get(field) == null) {
+			return;
+		}
+		ObjectMapper mapper = ObjectMapperFactory.jsonMapper();
+		mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+		ObjectNode parentObjNode = (ObjectNode) jsonNode;
+		parentObjNode.replace(field, mapper.valueToTree(updatedObject));
+	}
+
+	/**
+	 * Fetch json object based on {@link #getPath()} from service spec node
+	 * Converts json object to java object
+	 * @param serviceSpecObjNode
+	 * @param type
+	 * @return POJO, null if not found
+	 * @throws HyscaleException
+	 */
+	protected T getModel(ObjectNode serviceSpecObjNode, Class<T> type) throws HyscaleException {
+		if (serviceSpecObjNode == null) {
+			return null;
+		}
+		if (getPath() == null) {
+			return null;
+		}
+		JsonNode jsonNode = getNodeAtPath(serviceSpecObjNode, getPath());
+		if (jsonNode == null || jsonNode.isNull()) {
+			return null;
+		}
+		T t = null;
+		try {
+			t = mapper.treeToValue(jsonNode, type);
+		} catch (JsonProcessingException e) {
+			logger.error("Error while processing service spec ", e);
+			throw new HyscaleException(ControllerErrorCodes.SERVICE_SPEC_PROCESSING_FAILED, e.getMessage());
+		}
+
+		return t;
+	}
+
+	/**
+	 * 
+	 * @param parentObjectNode
+	 * @param path
+	 * @return JsonNode present at path, null if absent
+	 */
+	private JsonNode getNodeAtPath(ObjectNode parentObjectNode, String... path) {
+		JsonNode jsonNode = parentObjectNode;
+		for (int i = 0; i < path.length; i++) {
+			if (jsonNode == null || jsonNode.isNull()) {
+				return null;
+			}
+			jsonNode = jsonNode.get(path[i]);
+		}
+		return jsonNode;
+	}
+
+	/**
+	 * Implementing class returns object
+	 * 
+	 * @param serviceSpecObjNode
+	 * @return POJO
+	 * @throws HyscaleException
+	 */
+	protected abstract T getModel(ObjectNode serviceSpecObjNode) throws HyscaleException;
```

Comment:
a simple get() should work ?
getModel(objectNode, T) is more appropriate for its implementation